pkg/util/yaml_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+var (
+	repoRootDir string
+	testDataDir string
+)
+
+func init() {
+	wd, err := os.Getwd()
+	if err != nil {
+		panic(err)
+	}
+	repoRootDir = filepath.Join(wd, "../..")
+}
+
+func TestReadLayeredYAMLs(t *testing.T) {
```

Comment:
This test would also be more useful if it were testing the whole command that was using it. it would catch regressions, cmd parsing problems etc.